sched/sched/sched_releasetcb.c

```diff
@@ -163,7 +163,10 @@ int nxsched_release_tcb(FAR struct tcb_s *tcb, uint8_t ttype)
 
       /* And, finally, release the TCB itself */
 
-      kmm_free(tcb);
+      if ((tcb->flags & TCB_FLAG_PRESERVED_TCB) == 0)
```

Comment:
where the codd set this new flag

Reply:
from application who managed the tcb life cycle:

```
|  static struct task_tcb_s g_tcb;
|
|  memset(&g_tcb, 0, sizeof(struct task_tcb_s));
|  g_tcb.cmn.flags = TCB_FLAG_TTYPE_KERNEL | TCB_FLAG_PRESERVED_TCB;
|  nxtask_init(&g_tcb, "PTCB", 101, NULL, 1024, ptcb_task, NULL, NULL, NULL);
|
|  ...
|  nxtask_activate(&g_tcb.cmn);
```

Reply:
should we reverse the meaning (e.g. TCB_FLAG_FREE_TCB) and let's pthread_create/kthread_create/task_create set the new flag?

Reply:
Done